src/main/java/types/DisplayType.java

```diff
@@ -49,5 +50,49 @@ public enum DisplayType {
      * @date 23 Apr 2017
      * @status added
      */
-    SPICE;
+    @Deprecated
+    SPICE,
+
+    /**
+     * Display of type CIRRUS
+     * @author Jean-Louis Dupond <jean-louis@dupond.be>
+     * @date 26 May 2023
+     * @status added
+     */
+    @Deprecated
+    CIRRUS,
+
+    /**
+     * Display of type QXL
+     * @author Jean-Louis Dupond <jean-louis@dupond.be>
+     * @date 26 May 2023
+     * @status added
+     */
+    @Deprecated
+    QXL,
+
+    /**
+     * Display of type VGA
+     * @author Jean-Louis Dupond <jean-louis@dupond.be>
+     * @date 26 May 2023
+     * @status added
+     */
+    VGA,
+
+    /**
+     * Display of type BOCHS
+     * @author Jean-Louis Dupond <jean-louis@dupond.be>
+     * @date 26 May 2023
+     * @status added
+     */
+    BOCHS,
+
+  /**
+     * Display of type NONE
+     * This is for headless VM's
+     * @author Jean-Louis Dupond <jean-louis@dupond.be>
+     * @date 26 May 2023
+     * @status added
+     */
+    NONE;
```

Comment:
we have a long thread about this one back then - this couldn't work for specifying that we want a headless vm, we rather need to create the vm and then remove its video devices

Reply:
This is how it's represented now:
https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/DisplayType.java#L12

I think this should be fixable from the engine code.
Thing is that before this PR (and the ovirt-engine PR for this change), you couldn't set the DisplayType with the API, the engine code derived the DisplayType from the GraphicsType you passed.

Reply:
@sgratch is there a chance you can find the discussion we had with Juan back then?

Reply:
@dupondje @ahadas @liranr23 
Trying to restore my memory regarding the discussions and changes for supporting video device types and headless VMs via the rest API ended up with the following:

 1. Juan pushed to a solution of adding a VM 'headless' and a 'video type' attributes for supporting all combinations of video/graphics/headless via the API and we came with a long design thread for adding both, but at the end we decided not to invest on this and stayed with a partial  API support https://bugzilla.redhat.com/show_bug.cgi?id=1406394
 
 2. The solution we ended up with is was based on the old code, as described here: https://github.com/sgratch/ovirt-site/blob/master/source/develop/release-management/features/virt/headless-vm.html.md#rest-api, includes the following Main concept:
    - We can only set the graphics device type via the API.
    - The video device type logic is set on backend, based on supported OS combinations of video/graphics
    - For headless, we should remove all graphics console devices. This will set the video type to DisplayType.none on backend. 


3. For deprecating QXL and replacing it with VGA as the default video device for exising VMs, based on what I understand from https://bugzilla.redhat.com/show_bug.cgi?id=1976607, is that  the user is recommended to remove the graphic and video devices from the VM (creating a headless VM) and then adding a VNC graphic device. This can be done via REST in 2 phases of setting to headless and then setting to VNC.

4. The current API Model DisplayType values represents the graphics info. For supporting and setting the video display type as well, in addition to graphics type , you should as a 1st step to add a new attribute to the  VMs general info in addition to the graphics, as part of the [Display](http://ovirt.github.io/ovirt-engine-api-model/master/#types/display) attribute or as a new attribute under the VM info. Please note that backward compatibility should be kept so you can't just change an existing attribute values. Check the mappings https://github.com/oVirt/ovirt-engine/blob/504baab74e6b846a0cd1a593cd34f3665dbbedb7/backend/manager/modules/restapi/types/src/main/java/org/ovirt/engine/api/restapi/types/DisplayMapper.java#L96

Reply:
I remember we decided to have 2 phases to set the VM headless.
Based on [packaging/dbscripts/upgrade/04_05_0290_change_default_display.sql ](https://github.com/liranr23/ovirt-engine/blob/f45b601251f6add3a5dd11bc7582f0e1a1e8781b/packaging/dbscripts/upgrade/04_05_0290_change_default_display.sql), we change the default only for new VMs. I don't think we had a method to existing, other than defaulting in the UI to VGA. Therefore, for API ([DisplayMapper](https://github.com/oVirt/ovirt-engine/blob/master/backend/manager/modules/restapi/types/src/main/java/org/ovirt/engine/api/restapi/types/DisplayMapper.java)), - yes  I think you can switch to headless and then setting to VNC.
@sgratch point makes sense to me.